README.md

```diff
@@ -1,156 +1,65 @@
-* Creates `build-{pg_version}` folders with citus configured, so these folders have the necessary `Makefile.global` to run the tests. These folders are generated for each postgres version. For example, at the time of writing there are 3 folders with `build-10`, `build-11` and `build-12`.
+This will push all images, building all layers that might have changed since the last run of build. Make sure you have tested the images before pushing a release. CI might start using the newly pushed images directly, depending on the availability of a cache and how it is invalidated.
 
-The general process for other images are similar, which is:
+Before being able to push to the docker registry you need to have your cli authenticated to the docker hub and have sufficient privilidges to push to the registery.
 
-While building the image:
+If you don't have access, or want to push the images to a private repo, the repo can be changed at runtime with the `DOCKER_REPO` variable like:
 
-* Install postgres major which is taken as an argument `PG_MAJOR`.
+    $ DOCKER_REPO=private-repo make push-all
 
-While running the container:
+## 2. Images
 
-* Untar the citus tar to install citus for the postgres version that the image will use. For example, if postgres 11 is used, the tar is named `install-11.tar`.
-* The script (`install-and-test-ext`) takes the target name as an argument, and the following command is used to run the test:
+Details on the images. Mostly uninteresting for users. Please refer to the [Makefile](#1-makefile) section above.
 
-```bash
-gosu circleci make -C "${CIRCLE_WORKING_DIRECTORY}/build-${PG_MAJOR}/src/test/regress" "${@}"
-```
+### extbuilder
 
-* Finally `regression.diffs` is printed if it exists, which indicates that there was a problem during the tests.
+The [extbuilder](https://github.com/citusdata/the-process/tree/master/circleci/images/extbuilder) image is the first image that other jobs depend on in our tests. The [extbuilder](https://github.com/citusdata/the-process/tree/master/circleci/images/extbuilder):
 
-Because of the specific logic that we have to run tests faster, we have a docker image `debbuilder` that generates a debian package for `postgresql-server-dev-{pg-major}`. The generated package is in the built docker image, specifically in `{container-name}/home/circleci/debs`. The package is pushed to [the-process](https://packagecloud.io/citus-bot/the-process). This package place is added to apt source list and pinned with with the highest priority so that it is chosen over the standard `postgresql-server-dev-{pg-major}` package.
+This image contains all the artifacts required to produce a build of citus binaries for exactly 1 postgres version. This image is build for every supported Postgres version. Any scripts driving the build are contained in the citus repostiroy.
```

Comment:
build -> built

Reply:
which one of the 3? :P
built is past tense
build is current tense

Reply:
`This image is built `